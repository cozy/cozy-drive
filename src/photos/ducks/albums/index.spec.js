import React from 'react'
import { shallow } from 'enzyme'
import { AlbumPhotosWithLoader } from './index'

describe('Album view', () => {
  const AlbumPhotos = AlbumPhotosWithLoader({ children: null })

  it('should show a loader', () => {
    const component = shallow(<AlbumPhotos data={[]} fetchStatus="loading" />)
    expect(component).toMatchSnapshot()
  })

  it('should show an album when loaded', () => {
    const component = shallow(
      <AlbumPhotos
        data={{
          photos: {
            data: [],
            hasMore: false,
            fetchMore: jest.fn()
          }
        }}
        fetchStatus="loaded"
      />
    )
    expect(component).toMatchSnapshot()
  })

  it('should show a spinner when no album can be loaded', () => {
    const component = shallow(<AlbumPhotos data={null} fetchStatus="loaded" />)
    expect(component).toMatchSnapshot()
  })
})
