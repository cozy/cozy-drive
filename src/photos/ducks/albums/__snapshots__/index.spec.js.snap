// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Album view should show a loader 1`] = `
<withI18n(Loading)
  loadingType="photos_fetching"
  middle={true}
  size="xxlarge"
/>
`;

exports[`Album view should show a spinner when no album can be loaded 1`] = `
<withI18n(Loading)
  loadingType="photos_fetching"
  middle={true}
  size="xxlarge"
/>
`;

exports[`Album view should show an album when loaded 1`] = `
<withI18n(withRouter(Connect(AlbumPhotos)))
  album={
    Object {
      "photos": Object {
        "data": Array [],
        "fetchMore": [MockFunction],
        "hasMore": false,
      },
    }
  }
  fetchMore={[Function]}
  hasMore={false}
  photos={Array []}
/>
`;
